### DISCUSSION TITLE
Can I remove relay(?) tasmota creates with udisplay installation?

### DISCUSSION THREAD

**Discussion by catdogmaus** [arendst/Tasmota#22121] [Q&A] [answered]
I have not used display with tasmota before and it was really surprise for me that with display it creates also some kind of relay to turn it on/off.
The fact that I do not have any control over that is really annoying. Especially when I want to use rules. First the display was just `power` in console. When I added `relay1` it suddenly becomes `power2` when I will add some other relays it changes again. When I want to use rules that manipulate power relays it is virtually impossible to know do they function correctly. Also simply to have completely unnecessary button creates unnecessary confusion in GUI considering that it is not possible to rename them.
As I said this is really really annoying and I would like to remove that thing completely. Display should be on and that's it! 
Is there any command or option to make that possible?
thanks

**Comment by lalo-uy** [comment]
As was said many times the GUI is not the main way to handle tasmota.
The buttons can be renamed :
WebButton<x> Change the name of the toggle buttons of the WEB UI. This
command accepts spaces in the name

**Reply by sfromis** [reply]
You could also rename it with Unicode chars like: WebButton1 ⛔
And yes, basic Tasmota features in the rudimentary admin/test web UI are not open to detailed selection of what you'd like to see or not.

**Reply by catdogmaus** [reply]
I would not even mind to have that button when I had any control over it. The way at the moment how tasmota handles that is **really, really** bad. The screen on/off relay should have at least dedicated ID. At the moment it jumps around every time some other gpios are changed. I can of course rename it, lets say - `button2` to `screen` but now when I add (or remove) any other relay, button named screen is not suddenly screen anymore! Now the screen relay is actually `button3` or `button4`. As I mentioned same problems are with creating rules. What was `power1` or `power2` is suddenly not so anymore.

**Reply by sfromis** [reply, marked as answer]
When you reconfigure, you should not expect rules to stay the same. And no, you get no fixed power channel for a display, as it is deliberately placed last to avoid interfering with relays. In general, you cannot chose what numbers you'd want, as all those numbers are starting with 1, and counting up without skipping numbers.

**Reply by catdogmaus** [reply]
I have one question not directly related to display. 
I run rule that looks at 2 temp sensors and manipulates relay accordingly. (It is literally copy/paste from cookbook) I can see temperature changing but rule triggers extremely slow. When I say slow, it is minutes! TelePeriod is set to 10.

**Reply by sfromis** [reply]
Hard to say anything without seeing your specific rule set.

**Reply by catdogmaus** [reply]
1 is literally copy from cookbook
2 is workaround for manual switch with dummy relay
2 is working very well, 1 takes every time 2-3 minutes for triggering (both, on and/or  off).

```
mem3 30

rule1
  on DS18B20-1#temperature do
    event t1=%value%
  endon
  on DS18B20-2#temperature do
    event t2=%value%
  endon
  on event#t2>%mem3% do 
    var1 1;
  endon
  on event#t2<=%mem3% do 
    var1 0;
  endon
  on event#t1 do 
    backlog
    var2 %value%;
    add2 1;
  endon
  on event#t1 do 
    backlog
    var3 %value%;
    add3 2;
  endon
  on event#t2>%var3% do
    power1 %var1%;
  endon
  on event#t2<%var2% do
    power1 0;
  endon

rule1 1

----------------------------------

rule2
 on Power2#state==1 do rule1 0 endon
 on Power2#state==1 do power1 1 endon
 on Power2#state==0 do rule1 1 endon
 
rule2 1
```

**Reply by pkkrusty** [reply]
>   on event#t1 do 
>     backlog
>     var2 %value%;
>     add2 1;
>   endon

If I'm not mistaken, you can't perform two operations on a var in the same trigger. i.e. assign the value of t1 to var2 and then immediately perform math on it. You'd have to do that in a separate trigger. So...

on event#t1 do var2 %value% endon
on event#t1 do add2 1 endon

Those triggers are processed sequentially and should work.

**Reply by sfromis** [reply]
Actually, that specific example was ok, what you can't do is to first change a var in a rule, and then try to use it like `%var1%`, as these references are expanded before the whole rule starts.

**Reply by catdogmaus** [reply]
Meanwhile I did search for solution myself and it turns out that there are two completely different solutions for same example available.
https://github.com/tasmota/docs-7.1/blob/master/Rule-Cookbook.md#solar-heater-control is probably older version but comes out in google as first when search for rule cookbook.
https://tasmota.github.io/docs/Rules/#solar-heater-control Is more recent version and actually works without glitch.

**Comment by lalo-uy** [comment]
Adding new relay to a system is not somethin you do every week ;-)

**Reply by catdogmaus** [reply]
That's sure but I still have opinion that to have reserved dedicated power channel for display would avoid lots of confusion. It could be, lets say 28. Now its always last and always in same place. The goal here should be to make user experience to new users clear and minimally confusing as possible.
